tests/coverage/mcdc_non_control_flow.rs

```diff
@@ -0,0 +1,71 @@
+#![feature(coverage_attribute)]
+//@ edition: 2021
+//@ min-llvm-version: 18
+//@ compile-flags: -Zcoverage-options=mcdc
+//@ llvm-cov-flags: --show-mcdc
```

Comment:
It would be helpful to have a brief comment explaining what this test is testing.

Reply:
Thanks, will do !
For the record, this test ensures that boolean expressions that are not inside control flow decisions are correctly instrumented.